### ISSUE TITLE
[FEATURE] Logging frequency for TensorBoard

### ISSUE THREAD

**Issue by S-aiueo32** [williamFalcon/pytorch-lightning#156]
**Is your feature request related to a problem? Please describe.**
TensorBoard is a great tool for visualization but the size of event files rapidly grows bigger when we log images.
To prevent this problem, we need to set logging frequencies.
As the best of  my knowledge, in pytorch-lightning, we can manually set them like:
```python
class CoolModule(pl.LightningModule):
    def __init__(self, args):
        self.log_feq = args.log_freq

        self.model = ...

    def training_step(self, data_batch, batch_nb):
        input = data_batch['input']
        output = self.forward(input)

        if self.global_step % self.log_freq == 0:
            self.experiment.add_image('output_image', output, self.global_step)
```
This is an easy way, however, I think it is clearer to control frequencies by `Trainer`.

**Describe the solution you'd like**
Add an option for controlling the frequencies to `Trainer`.
```python
trainer = Trainer(model, tb_log_freq=foo)
```
To enable this functionality, `training_step` should return image tensors like:
```python
def training_step(self, data_batch, batch_nb):
    input = data_batch['input']
    output = self.forward(input)
    loss = ...
    return {'loss', loss, 'prog': {'loss': loss}, 'image': {'output': output}}
```

**Comment by williamFalcon** [comment]
already exists. maybe the flag can be renamed for clarity?

https://williamfalcon.github.io/pytorch-lightning/Trainer/Logging/#log-metric-row-every-k-batches

**Comment by williamFalcon** [comment]
the second case:  
```
        if self.global_step % self.log_freq == 0:
            self.experiment.add_image('output_image', output, self.global_step)
```   

does have to be done manually. However, you can use the flag ```self.trainer.add_log_row_interval``` instead of creating your own

**Comment by S-aiueo32** [comment]
Thank you for replying. I missed the option, sorry.
It may be better to add the description to Logging/TensorBoard section, rather than renaming.

**Comment by S-aiueo32** [comment]
And uh, do you have any plans to manage the other information(e.g. images and histograms) through Trainer?
With a little change around [trainer.py#L887](https://github.com/williamFalcon/pytorch-lightning/blob/master/pytorch_lightning/models/trainer.py#L887), it can be true easily, I think.

**Comment by williamFalcon** [comment]
@S-aiueo32 we thought about it, but it's better to defer logging to the user.  We 're looking to support other loggers as well, in which case the user should decide what to do with it.

If  you want to contribute, that's a good  ticket to take #111
